process/transaction/interceptedTransaction.go

```diff
@@ -192,6 +201,8 @@ func (inTx *InterceptedTransaction) verifySig() error {
 		return err
 	}
 
+	inTx.whitelistVerified.Add([][]byte{inTx.Hash()})
```

Comment:
We are continuously adding in the cache verified transactions until the cache is full. This will degrade the performance when checking the transaction signatures as there is an extra step: search through all 50000 hashes to check if it is contained (and then, most probably, do the signature check). Suggestion to add a remove hash call on an `else` branch on line 133 as to keep the transactions count in the cache at a minimum.

Reply:
added the remove on else

Reply:
in main a disabled component is created if the node is not observer.